### DISCUSSION TITLE
So, what's the difference between ptg and textual?

### DISCUSSION THREAD

**Discussion by longqzh** [bczsalba/pytermgui#104] [General]
So, what's the difference between ptg and textual?

**Comment by bczsalba** [comment]
A lot!

Textual aims for a web-like experience, full of things like Async and web habits (such as a custom `Reactive` class vs. a standard property. PyTermGUI tries to be as Pythonic as absolutely possible, using a bunch of Python specific features and syntacies to make your life (and code) a lot nicer. I personally don't like working in the web-dev world, so in most places I try to look for inspiration in traditional, native UI frameworks (such as QT) rather than towards HTML and CSS.

Neither of these are better than the other, they are just different approaches to the same end goal. Both can make mostly the same interfaces with the same interactions, but take relatively different steps to get there.

Another important thing to mention is the development team's size. Textual (under Textualize) has become a funded project, with the goal of eventually making a sort of Python to web-app transpiler AFAIK. Due to this funding they are able to afford an office with multiple, highly skilled and specialized employees. PTG on the other hand is 'but one dude, me! I literally started programming around 3 years ago, so my experience is nowhere near any of those guys. I also have a job where I work close to full-time hours, next to going to university. In effect, this means that I have a fraction of the man-hours to dedicate to the project, sadly.

Despite all of that, I think I've made a pretty nifty library. It has some rough edges, but I'm working away to sand those up in a timely fashion. Textual is already a more "complete" library in some aspects, but that is unfortunately unavoidable with the circumstances described above. Don't get me wrong; PTG is **completely capable** of building cool and complex UI, and most of the upcoming updates are things any random person could implement on top of already existing elements.

Take all of that how you will. I recommend playing around with both libraries, seeing what works and what doesn't and making your own choice accordingly. I personally prefer this one, but I see the pull of Textual (or any of the often unsung alternatives, such as urwid or PicoTUI, next to the hundreds of projects in other languages, such as those made by [Charm](https://github.com/charmbracelet)).

I do hope you stick around, however! :)
